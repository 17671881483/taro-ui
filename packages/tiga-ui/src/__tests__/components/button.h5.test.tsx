import React, { Fragment } from 'react'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'

describe('AtButton test in h5', () => {
  const OLD_ENV = { ...process.env }
  process.env.TARO_ENV = 'h5'

  afterAll(() => {
    process.env = { ...OLD_ENV } // Restore old environment
    jest.resetModules()
  })

  afterEach(cleanup)

  const AtButton = jest.requireActual('../../components/button').default

  it('render by default', () => {
    const component = renderer.create(<AtButton>click me</AtButton>)

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render by different type', () => {
    const component = renderer.create(
      <Fragment>
        <AtButton type='primary'>primary</AtButton>
        <AtButton type='secondary'>secondary</AtButton>
        <AtButton type='tertiary'>tertiary</AtButton>
      </Fragment>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('render by different size', () => {
    const component = renderer.create(
      <Fragment>
        <AtButton size='full'>full</AtButton>
        <AtButton size='large'>large</AtButton>
        <AtButton size='medium'>medium</AtButton>
        <AtButton size='mini'>mini</AtButton>
        <AtButton size='small'>small</AtButton>
        <AtButton size='tiny'>tiny</AtButton>
      </Fragment>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('when fire click clickFn should be called', () => {
    const clickFn = jest.fn()
    const { getByText } = render(
      <AtButton onClick={clickFn} size='tiny'>
        click
      </AtButton>
    )

    fireEvent.click(getByText(/click/i))
    expect(clickFn).toBeCalledTimes(1)
  })

  it('when fire click a disabled button clickFn should not be called', () => {
    const clickFn = jest.fn()
    const { getByText } = render(
      <AtButton disabled onClick={clickFn}>
        click
      </AtButton>
    )

    fireEvent.click(getByText(/click/i))
    expect(clickFn).toBeCalledTimes(0)
  })
})
