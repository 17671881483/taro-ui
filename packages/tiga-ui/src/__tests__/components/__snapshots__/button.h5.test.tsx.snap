// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtButton test in h5 render by default 1`] = `
<view
  className="at-button at-button--secondary at-button--medium"
  onClick={[Function]}
>
  <button
    className="at-button__wxbutton"
    lang="zh_CN"
  />
  <view
    className="at-button--secondary__text at-button--medium__text"
  >
    click me
  </view>
</view>
`;

exports[`AtButton test in h5 render by different size 1`] = `
Array [
  <view
    className="at-button at-button--secondary at-button--full"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--full__text"
    >
      full
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--large"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--large__text"
    >
      large
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--medium"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--medium__text"
    >
      medium
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--mini"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--mini__text"
    >
      mini
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--small"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--small__text"
    >
      small
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--tiny"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--tiny__text"
    >
      tiny
    </view>
  </view>,
]
`;

exports[`AtButton test in h5 render by different type 1`] = `
Array [
  <view
    className="at-button at-button--primary at-button--medium"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--primary__text at-button--medium__text"
    >
      primary
    </view>
  </view>,
  <view
    className="at-button at-button--secondary at-button--medium"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--secondary__text at-button--medium__text"
    >
      secondary
    </view>
  </view>,
  <view
    className="at-button at-button--tertiary at-button--medium"
    onClick={[Function]}
  >
    <button
      className="at-button__wxbutton"
      lang="zh_CN"
    />
    <view
      className="at-button--tertiary__text at-button--medium__text"
    >
      tertiary
    </view>
  </view>,
]
`;
